src/main/java/dev/enjarai/trickster/data/conversion/BlockConversionLoader.java

```diff
@@ -99,6 +102,17 @@ public boolean convert(Block block, World world, BlockPos pos) {
             world.syncWorldEvent(WorldEvents.BLOCK_BROKEN, pos, Block.getRawIdFromState(blockState));
         }
 
+        if (world.getDimension().ultrawarm()) {
+            if (blockState.isOf(Blocks.WATER)) {
+                blockState = Blocks.AIR.getDefaultState();
+            } else if (blockState.getFluidState().isOf(Fluids.WATER)) {
+                Optional<Boolean> perhapsWaterlogged = blockState.getOrEmpty(Properties.WATERLOGGED);
+                if (perhapsWaterlogged.isPresent() && perhapsWaterlogged.get()) {
+                    blockState.with(Properties.WATERLOGGED, false);
```

Comment:
What exactly does this line do? BlockStates are immutable, so this wont update the `blockState` variable right?

Reply:
oh, oops